doc/nrf/releases_and_maturity/migration/migration_3.1_54h_suit_ironside.rst

```diff
@@ -0,0 +1,173 @@
+          memory-region = <&cpusec_cpuapp_ipc>;
+          status = "okay";
+      };
+
+Update PERIPHCONF
+-----------------
+
+The new UICR format no longer holds peripheral configuration initial values.
+You must generate a PERIPHCONF blob at build time.
+
+The Zephyr build invokes the :file:`gen_uicr.py` script (:file:`soc/nordic/common/uicr/gen_uicr.py` in the Zephyr tree) via ``nrf-regtool`` in the NCS sysbuild system.
+When the following Kconfig options are set:
+
+  * :kconfig:option:`CONFIG_NRF_HALTIUM_GENERATE_UICR` to ``y``
+  * :kconfig:option:`CONFIG_NRF_HALTIUM_UICR_PERIPHCONF` to ``y``
+
+The script does the following:
+
+  1. It reads the ``peripconf_partition`` node in the devicetree to discover the partition's address and size.
+  #. It extracts the address/value pairs from the ``PERIPHCONF`` section of the Zephyr ELF image.
+  #. It generates two Intel HEX files:
+
+    * :file:`uicr.hex` - the new UICR entries
+    * :file:`periphconf.hex` - the MRAM-resident ``PERIPHCONF`` blob
+
+Both HEX files must be flashed alongside your firmware image.
+``west flash`` handles this automatically.
+
+You do not need to modify your application code.
+You only need to ensure the DTS partition exists.
+
+Memory protection and isolation
+-------------------------------
+
+IronSide SE currently grants full memory-access permissions to both application and radio domains by default.
+Delete any UICR settings related to the following:
+
+* Secure Domain IPC buffer location
+* Secure-Domain offsets
+* Partition lock bits
+
+Memory map changes
+------------------
+
+With IronSide SE, the memory map changed as follows:
+
+* The application core firmware now always starts at address ``0xE03_0000``, which is the first address in ``MRAM00`` immediately following the IronSide firmware.
+* Nordic-reserved partitions in ``MRAM11`` and ``RAM0x`` have been removed.
+* IPC buffers toward the Secure Domain are relocated to fixed addresses in ``RAM20``.
+  Memory previously reserved in ``RAM0x`` for IPC can now be repurposed.
+* The devicetree no longer uses the ``nordic,owned-memory`` or ``nordic,owned-partitions`` compatibles.
+  Refer to the `nRF54H20 DK memory map`_ for details.
+
+To enable ``UICR/PERIPHCONF`` generation, ensure a DTS partition labeled ``peripconf_partition`` exists with sufficient size (for example, 8 KBs) to embed the generated address-value blob.
+Move all partition definitions directly under the ``mram1x`` partitions node.
+
+DFU support with MCUboot
+------------------------
+
+IronSide SE drops SUIT in favor of MCUboot.
+To migrate DFU, do the following:
+
+* Remove SUIT-specific Kconfig symbols from both :file:`prj.conf` and :file:`sysbuild.conf` files.
+* Delete SUIT manifest templates, typically located in the :file:`suit` directory.
+* Choose one of the supported MCUboot bootloader modes.
+* Rename ``dfu_partition`` to ``cpuapp_slot1_partition``, or remove it if your chosen mode does not require a DFU slot.
```

Comment:
Unfortunately, it is more complicated.
Actually:
 - If your chosen mode does not require a DFU slot, remove the `dfu_partition`. Otherwise split the `dfu_partition` into `cpuapp_slot1_partition` and `cpurad_slot1_partition1`. Those partitions should have the same size as their counterpart (`cpuapp_slot0_partition` and `cpurad_slot0_partition` respectively).